eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -129,6 +149,71 @@ emptyGenesisStaking =
       sgsStake = mempty
     }
 
+-- | Unlike @'NominalDiffTime'@ that supports @'Pico'@ precision, this type
+-- only supports @'Micro'@ precision.
+newtype NominalDiffTimeMicro = NominalDiffTimeMicro Micro
+  deriving (Show, Eq, Generic)
+  deriving anyclass (NoThunks)
+  deriving newtype (Ord, Num, Fractional, Real, ToJSON, FromJSON)
+
+deriving instance Generic Micro
+
+deriving anyclass instance NoThunks Micro
```

Comment:
It is better to move this instance into `Cardano.Ledger.Orphans` and leave  a `TODO` that we should upstream instance `HasResolution p => NoThunks (Fixed p)` into `nothunks` package
```suggestion
deriving anyclass instance NoThunks Micro
```

Reply:
Awesome! Thanks! :bow: